File: 24 Swap Nodes in Pairs/swapPairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        
        ListNode* dummyHead = new ListNode(0);
        dummyHead->next = head;
        ListNode* p = dummyHead;
        
        while( p->next && p->next->next ){
            
            ListNode* node1 = p->next;
            ListNode* node2 = node1->next;
            ListNode* next = node2->next;
            
            node2->next = node1;
            node1->next = next;
            p->next = node2;
            
            p = node1;
        }
        
        ListNode* retNode = dummyHead->next;
        delete dummyHead;
        
        return retNode;
    }
};

File: 24 Swap Nodes in Pairs/swapPairs.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        
        dummyHead = ListNode(0)
        dummyHead.next = head
        p = dummyHead
        
        while p.next != None and p.next.next != None:
            
            node1 = p.next
            node2 = node1.next
            nextOne = node2.next
            
            node2.next = node1
            node1.next = nextOne
            p.next = node2
            
            p = node1
            
        retNode = dummyHead.next
        dummyHead.next = None
        
        return retNode